### DISCUSSION TITLE
是否考虑增加固定导航栏navbar的配置

### DISCUSSION THREAD

**Discussion by YiSunYiMao** [ppoffice/hexo-theme-icarus#1038] [Ideas]
类似目录的固定

**Comment by Artist-x** [comment]
在css/style.styl中添加以下代码、或单独新建一个自定义.styl文件并在style.styl中引用它！

.navbar-main
    position: fixed !important
    width: 100% !important
    top: 0px

.section
    margin-top: 45px

**Reply by JIeJaitt** [reply]
如果使用 npm 安装的，这段代码添加在哪儿？

**Comment by PJ-568** [comment]
> @JIeJaitt 
>
> 如果使用 npm 安装的，这段代码添加在哪儿？

node_modules/hexo-theme-icarus/source/css

**Reply by JIeJaitt** [reply]
我用hexo注入器的方式做到了，这种做法只适合本地打包好部署到github的，如果是gtihub action自动部署的那么就没有node_modules这个目录

https://github.com/ppoffice/hexo-theme-icarus/discussions/1218

**Reply by PJ-568** [reply]
受教了。我一直是保存了修改过后的文件拷贝，在 Github Actions 安装好依赖后覆盖到 `node_modules/hexo-theme-icarus/source/css`。

**Reply by JIeJaitt** [reply]
不客气，请问博主你的blog右上角的语言切换功能这么做的？

**Reply by PJ-568** [reply]
我引入了 [translate.js](//github.com/xnx3/translate) 来实现前端翻译。详见[这篇文章](https://pj568.eu.org/2023/09-23-Rebuild_Hexo_site/)的`配置自动翻译`部分。

**Reply by JIeJaitt** [reply]
哦哦好的，感谢
